streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -149,6 +153,23 @@ public Integer apply(Integer value1, Integer value2) {
     }
 
     @Test
+    public void testTimestampExtractorsWithThrough() {
+        final KStreamBuilder builder = new KStreamBuilder();
+        KStream<String, String> stream1 = builder.stream(stringSerde, stringSerde, "topic-1", "topic-2");
+        KStream<String, String> stream2 = builder.stream(stringSerde, stringSerde, "topic-3", "topic-4");
+
+        stream1.to("topic-5");
+        stream2.through("topic-6");
```

Comment:
Maybe we can extend this test, such that an internal repartitioning topic in created?